olive/passes/olive_pass.py

```diff
@@ -338,7 +339,15 @@ def _default_config(cls, accelerator_spec: AcceleratorSpec) -> Dict[str, PassCon
     def _resolve_defaults(cls, config: Dict[str, Any], default_config: Dict[str, PassConfigParam]) -> Dict[str, Any]:
         """Resolve default values."""
         for key, value in config.items():
-            if value == PassParamDefault.DEFAULT_VALUE:
+            if isinstance(default_config[key].searchable_values, CategoricalValue):
+                # Replace CategoricalValue with Categorical parameter
+                v = value
+                if v is None or isinstance(v, CategoricalValue):
```

Comment:
Is this check flipped?
When do you want to override the value with default_value?

Reply:
Override the value with default if the user hasn't provided an explicit value for the parameter as part of the config.

This is correct. In the above case, the value is of type `CategoricalValue`.